tests/console/validate_selected_patterns.pm

```diff
@@ -0,0 +1,36 @@
+# SUSE's openQA tests
+#
+# Copyright © 2019 SUSE LLC
+#
+# Copying and distribution of this file, with or without modification,
+# are permitted in any medium without royalty provided the copyright
+# notice and this notice are preserved. This file is offered as-is,
+# without any warranty.
+
+# Summary: Validation module to check patterns installed.
+# Scenarios covered:
+# - zypper search for named patterns(base,enhanced_base;
+# - zypper search for removed pattern apparmor
+#
+# Maintainer: Yiannis Bonatakis <ybonatakis@suse.de>
+
+use strict;
+use warnings;
+use base "opensusebasetest";
+use testapi;
+use utils;
+
+sub run {
+
+    select_console('root-console');
+
+    my $installed_patterns = get_var('PATTERNS');
+    record_info("Verify $installed_patterns");
+    zypper_call("se -x -t pattern -i -n base");
```

Comment:
Design of this module looks like we can reuse it to validate patterns in other installations too. What I find confusing that we read `PATTERNS` variable here, but don't really use it. What's about setting patterns as test data?

Reply:
Yes, the idea was to use the PATTERNS variables to loop through. but the variable doesnt correspond to the actual names. So i chose to keep it simple and i just display the info coming from the variable for visual verification in case that something goes wrong. 

Otherwise i should somehow map the variables from the PATTERNS to the actual registered names.

Reply:
I guess it would be better if we set relevant names in the variable. Question is if some of the installed patterns are dependencies of the ones we explicitly select, or they are all installed unconditionally. Then we might have mapping in the code or not validate those as we don't touch them.

Reply:
Checking the test, I would say that there are not dependencies. The minimal and base seems to be the one selected in the select_patterns_and_packages and the only ones in the zypper's query about the pattern. I am not sure if you have something else in your mind that i dont know.

Reply:
I would validate only pattern we select. Not sure where the other patterns are coming from.
What is confusing is that we read variable and state that we validate these patterns, but then validate some hardcoded list instead, so if someone changes variable, message states that we validate all these patterns, but we don't. So either way, let's make it consistent.

As it's first test suite where we do this, we are not sure how to use it, so we can start with hardcoded strings, but then we should not read the variable.

Reply:
I'm missing a loop here too @b10n1k, you can organize yaml data as you wish to accomplish that. It would be useful for validating pattern that should not appear as well, are you planning to add some exclusion list for minimal_x?, it should be some obvious patterns that a minimal_x system should not have, right?

Reply:
I try to address the comments on effaaaef3